frontend/app/src/components/home/admin/ReviewTranslationCorrections.svelte

```diff
@@ -15,27 +30,57 @@
     let verifications: Record<string, string> = {};
 
     onMount(() => {
-        client.getTranslationCorrections().then((res) => {
+        client.getProposedTranslationCorrections().then((res) => {
             corrections = flattenCorrections(res);
         });
     });
 
-    function flattenCorrections(corrections: TranslationCorrections): TranslationCorrection[] {
-        return Object.values(corrections)
-            .flatMap((val) => Object.values(val))
-            .filter((c) => !c.approved);
+    function flattenCorrections(corrections: CandidateTranslations[]): TranslationCorrection[] {
+        return corrections.flatMap((group) =>
+            group.candidates.map((correction) => ({
+                id: correction.id,
+                locale: group.locale,
+                key: group.key,
+                value: correction.value,
+                proposedBy: correction.proposedBy,
+                proposedAt: correction.proposedAt,
+            })),
+        );
     }
 
-    function approveCorrection(correction: TranslationCorrection) {
-        client
-            .approveTranslationCorrection(correction)
-            .then((res) => (corrections = flattenCorrections(res)));
+    function removeCorrection(id: bigint) {
+        corrections = corrections.filter((c) => c.id !== id);
     }
 
-    function rejectCorrection(correction: TranslationCorrection) {
-        client
-            .rejectTranslationCorrection(correction)
-            .then((res) => (corrections = flattenCorrections(res)));
+    function approveCorrection({ id }: TranslationCorrection) {
+        client.approveTranslationCorrection(id).then((success) => {
+            if (success) {
+                removeCorrection(id);
+            } else {
+                toastStore.showFailureToast(
+                    i18nKey("Sorry we were unable to approve this correction"),
+                );
+            }
+        });
+    }
+
+    function rejectCorrection({ id }: TranslationCorrection, reason: RejectReason) {
+        client.rejectTranslationCorrection(id, reason).then((success) => {
+            if (success) {
+                removeCorrection(id);
+            } else {
+                toastStore.showFailureToast(
+                    i18nKey("Sorry we were unable to reject this correction"),
+                );
+            }
+        });
+    }
+
+    function previewCorrection(correction: TranslationCorrection) {
+        // This will pretend that the value is english and apply it to the english i18n dictionary temporarily.
+        // This is just so that we have the option to look at it in the UI to check for layout problems
+        // client.previewTranslationCorrection({ ...correction, locale: "en" });
+        console.log("preview: ", correction);
```

Comment:
Is this supposed to be commented out?

Reply:
Uhm no I think I've forgotten to fill that in.